bin/gitlab-api-v3

```diff
@@ -103,42 +103,18 @@ try {
 
     exit 0 if !defined $data;
 
-    # Replace all non HASH/ARRAY references with strings.  This is
-    # implemented generically, but really only causes JSON boolean
-    # object to be stringified which makes sense for non-JSON serializers.
-    if ($format ne 'JSON') {
-        $data = clean_data( $data );
```

Comment:
`clean_data` might actually not be the source of the problem, now that I think about it. As long as the call to the serialiser is only applied to references (instead of to all responses), the `clean_data` function would be fine.

Maybe you'd prefer something like this?

```perl
if (ref $data) {
  print $serializer->serialize( clean_data $data );
}
else {
  print $data;
}
```